### ISSUE TITLE
Request mark isn't saved when marking a request from the pinned hosts and going back to Remote devices

### ISSUE THREAD

**Issue by AdamSafedk** [ProxymanApp/Proxyman#1351]
## Description
Pinned filters don't behave the same as the cmd+f filters

## Current Behavior
In 3.9.0 there was a huge improvement in the behavior of the filter, now if you choose/mark a request and remove your filter it still marks the request. **BUT**  the pinned hosts filter have still the old behavior, which means that if you choose/mark a request from the pinned hosts and then browse to the "Remote devices"  it doesn't save your choice.
## Steps to Reproduce

1. Choose one of you're pinned host
2. Mark/Choose a request
3. Go back to the Remote devices tab 
Notice that your marked request isn't marked anymore

## Environment

- App version: e.g Proxyman 3.9.0
- macOS version: e.g macOS Monterey

**Comment by NghiaTranUIT** [comment]
You're right. Currently, when the data source is changed (Clicking on the item on the Left Panel), the table view is completely reloaded. Thus, the selection is reset to the first row.

I will add this behavior soon 👍
